### PR TITLE
:sparkles: Cluster API v1alpha3

### PR DESCRIPTION
**Pull request by liztio**
**What this PR does / why we need it**:
Bumps the tests to cluster api v1alpha3

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -427,6 +427,8 @@ modernc.org/strutil v1.0.0/go.mod h1:lstksw84oURvj9y3tn8lGvRxyRC1S2+g5uuIzNfIOBs
 modernc.org/xc v1.0.0/go.mod h1:mRNCo0bvLjGhHO9WsyuKVU4q0ceiDDDoEeWDJHrNx8I=
 sigs.k8s.io/cluster-api v0.2.5 h1:N3nz2U62tE1lm4szCrEDMPMg/8RMiLaYgHWZJqkSuLE=
 sigs.k8s.io/cluster-api v0.2.5/go.mod h1:Agc72Ra5LMOkQQ2v/Ywv1KUemaYAwvkQ+G59Ym5H8e4=
+sigs.k8s.io/cluster-api v0.2.6-0.20191014202550-20da46b4c73c h1:Ccf5cuxf9pS/z/RtFvPCoRGM8BqHVK8a3FlqzBMr9nQ=
+sigs.k8s.io/cluster-api v0.2.6-0.20191014202550-20da46b4c73c/go.mod h1:MLM2W2O1cp8ui/Joyxg4U7NH2R8vZIRSzoR8/fVg2tA=
```
Comment (vincepri) [inline go.sum]: Seems like we might need a replace directive, given that's trying to import both

Comment (ncdc) [reply go.sum]: Or, now that cabpk is merged in, update the refs to its packages and that should eliminate the capi 0.2.5 dep, right?

Comment (vincepri) [reply go.sum]: hold on to that, in the current state cabpk is a sub-gomodule, which we shouldn't use

**Comment by liztio** [comment]
```
pkg/apis/awsprovider/v1alpha1/register.go:30:2: unknown import path "sigs.k8s.io/cluster-api/pkg/apis/deprecated/v1alpha1": cannot find module providing package sigs.k8s.io/cluster-api/pkg/apis/deprecated/v1alpha1
```

guess we can't yet :(

**Pull request merged**